### PR TITLE
[APPSRE-6069] add kinesis-to-opensearch lambda

### PR DESCRIPTION
**Pull request by BumbleFeng**
Introducing kinesis-to-opensearch lambda

along with the new get_lambda_zip function(will replace get_logtoes_zip and get_rosa_authenticator_zip in a following PR)

depends on https://github.com/app-sre/qontract-schemas/pull/230

Signed-off-by: Feng Huang <fehuang@redhat.com>

### CODE REVIEW

**Comment by BumbleFeng** [comment]
[test]

File: reconcile/utils/terrascript_aws_client.py [hunk 1]
```diff
@@ -415,6 +420,31 @@ def state_bucket_for_account(
...
+                # this may have already happened, so we check again
+                if not self.lambda_zip.get(release_url):
+                    self.lambda_zip[release_url] = self.download_lambda_zip(release_url)
+        return self.lambda_zip[release_url]
+
+    def download_lambda_zip(self, release_url: str) -> str:
+        if not self.token:
+            self.token = get_default_config()["token"]
+        headers = {"Authorization": "token " + self.token}
+        r = requests.get(GH_BASE_URL + "/" + release_url, headers=headers, timeout=60)
```
Comment (esron) [inline reconcile/utils/terrascript_aws_client.py:435]: Couldn't this be a case to extend RawGithubApi?

File: reconcile/utils/terrascript_aws_client.py [hunk 2]
```diff
@@ -166,6 +167,9 @@
 
 GH_BASE_URL = os.environ.get("GITHUB_API", "https://api.github.com")
 LOGTOES_RELEASE = "repos/app-sre/logs-to-elasticsearch-lambda/releases/latest"
+KINESISTOOS_RELEASE = (
```
Comment (rporres) [inline reconcile/utils/terrascript_aws_client.py:170]:
```suggestion
KINESIS_TO_OS_RELEASE = (
```

**Review by rporres — APPROVED** [review]

**Pull request merged**